### ISSUE TITLE
File with suffix of ".client.ts" causes compile error

### ISSUE THREAD

**Issue by hughanderson4** [roblox-ts/roblox-ts#1167]
## Version
**roblox-ts:** 1.0.0-beta.6

## Description
If I have a file with a name of `something.client.ts` it cannot be imported by a client script.

## Minimal Reproduction
1. Inside the default project structure, under `src\client\` create a file named `something.client.ts` and have something to export there like `export const testValue = 'a';`
2. Inside of a different file in your `src\client` folder attempt to reference the testValue
3. Linter will draw "red squiggly lines", click them and then use VS Code "lightbulb" to auto-import which will add at the top of your file like this: `import { someValue } from "./something.client";`
4. Now try to run `rbxtsc` and notice an error message like:

> src/client/someScript.ts:3:27 - error TS roblox-ts: Cannot import a non-ModuleScript!

Now rename the file from step 1 to `somethingClient.ts` and update the imports.
Notice the error goes away, based on the name of the file.

## Background
I started to adopt the naming convention of `.client.ts` and `.server.ts` suffixes based on some readings of [Kampfkarren code](https://github.com/Kampfkarren/zombie-strike/tree/master/src/mission/StarterPlayer/StarterPlayerScripts) and I think it's a nice pattern, especially with many tabs open in VS Code at the same time.

**Comment by Fireboltofdeath** [comment]
.client.ts syncs into a client script, which isn't a module and cannot be required normally in Roblox. If you need to import something from a script, you should make it a module.

**Comment by hughanderson4** [comment]
Hey thanks for the swift reply.  As I noted, this seems to work fine with a different filename.  It seems as though I triggered some special behavior by using that suffix.

**Comment by hughanderson4** [comment]
Actually this is a super-minimalistic breakdown of what's actually going on.  In my real solution I have an abstract base class defined in file named like base.client.ts which works fine to import from elsewhere in client folder, if the file is renamed to baseClient.ts

**Comment by Fireboltofdeath** [comment]
The type of script that Rojo syncs into studio is based on the file extension.
name.client.ts -> [LocalScript](https://developer.roblox.com/en-us/api-reference/class/LocalScript) name
name.server.ts -> [Script](https://developer.roblox.com/en-us/api-reference/class/Script) name
name.ts -> [ModuleScript](https://developer.roblox.com/en-us/api-reference/class/ModuleScript) name

The suffix you're using tells Rojo what to sync the script as (and by-proxy, roblox-ts) Roblox doesn't allow you to require LocalScripts (or normal Scripts) which is what you're telling Rojo to sync the file as.

**Comment by hughanderson4** [comment]
Totally makes sense, given that explanation.
And your words, triggered something in back of my mind, which remembered somewhere I already knew that, somehow.
I searched bunch of pages just now, and I finally found, where I must have read that before:
https://rojo.space/docs/0.5.x/reference/sync-details/
Here is the naming convention I have stumbled upon lol
Thanks so much for the support!  `rbxtsc` is a very awesome tool and is helping me out in tremendous ways.
